apps/condo/domains/common/utils/serverSchema/index.js

```diff
@@ -96,19 +96,30 @@ const loadListByChunks = async ({
     list,
     where = {},
     sortBy = ['createdAt_ASC'],
-    chunkSize = 100,
+    _chunkSize = 100,
     limit = 100000,
 }) => {
+    let chunkSize = _chunkSize <= limit ? _chunkSize : limit
+
+    if (chunkSize < 1 || limit < 1) throw new Error('Both chunkSize and limit should be > 0')
+    if (chunkSize > 100) chunkSize = 100
+
     let skip = 0
-    let maxiterationsCount = Math.ceil(limit / chunkSize)
-    let newchunk = []
-    let all = []
+    let iterations = Math.ceil(limit / chunkSize)
+    let result = []
+
     do {
-        newchunk = await list.getAll(context, where, { sortBy, first: chunkSize, skip: skip })
-        all = all.concat(newchunk)
-        skip += newchunk.length
-    } while (--maxiterationsCount > 0 && newchunk.length)
-    return all
+        const chunk = await list.getAll(context, where, { sortBy, first: chunkSize, skip: skip })
+
+        if (isEmpty(chunk)) break
+
+        result = result.concat(chunk)
+        skip += chunk.length
+
+        if (chunk.length < chunkSize) break
+    } while (--iterations > 0)
+
+    return result
```

Comment:
please rollback your changes the previous variant is shorter than yours and clean

Reply:
done